tool_water_balance/tests/test_waterbalance.py

```diff
@@ -1,49 +1,121 @@
+from pathlib import Path
 from qgis.core import QgsCoordinateReferenceSystem
 from qgis.core import QgsCoordinateTransform
 from qgis.core import QgsGeometry
-from qgis.core import QgsPoint
+from qgis.core import QgsPointXY
 from qgis.core import QgsProject
+from ThreeDiToolbox.tests.test_init import TEST_DATA_DIR
+from ThreeDiToolbox.tests.utilities import ensure_qgis_app_is_initialized
 from ThreeDiToolbox.tool_result_selection.models import TimeseriesDatasourceModel
 from ThreeDiToolbox.tool_water_balance.tools.waterbalance import WaterBalanceCalculation
+from ThreeDiToolbox.tool_water_balance.views.waterbalance_widget import BarManager
+from ThreeDiToolbox.tool_water_balance.views.waterbalance_widget import INPUT_SERIES
+from ThreeDiToolbox.tool_water_balance.views.waterbalance_widget import (
+    WaterBalanceWidget,
+)
 
-import os.path
-import unittest
+import mock
+import numpy as np
+import pytest
 
 
-test_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)))
+TESTMODEL_DIR = TEST_DATA_DIR.joinpath("testmodel", "v2_bergermeer")
+MODEL_SQLITE_PATH = TESTMODEL_DIR / "v2_bergermeer.sqlite"
+assert Path.is_dir(TESTMODEL_DIR), "testmodel dir not found"
+assert Path.is_file(MODEL_SQLITE_PATH), "modelspatialite not found"
 
 
-@unittest.skip("TODO: add the required files")
-class WaterbalanceClassTest(unittest.TestCase):
-    """Test the QGIS Environment"""
+# links_expected consists of (lines, pumps)
+LINKS_EXPECTED = (
+    {
+        "1d_in": [29136],
+        "1d_out": [],
+        "1d_bound_in": [],
+        "1d_bound_out": [],
+        "2d_in": [1557, 3689, 3690, 3710, 4879, 6573, 8900, 8901, 8964, 10314],
+        "2d_out": [3730, 4895, 5348, 10347, 10369],
+        "2d_bound_in": [],
+        "2d_bound_out": [],
+        "1d__1d_2d_flow": [],
+        "2d__1d_2d_flow": [],
+        "1d_2d_exch": [30649, 31654],
+        "2d_groundwater_in": [
+            17968,
+            20100,
+            20101,
+            20121,
+            21290,
+            22984,
+            25311,
+            25312,
+            25375,
+            26725,
+        ],
+        "2d_groundwater_out": [20141, 21306, 21759, 26758, 26780],
+        "2d_vertical_infiltration": [11591, 11592, 12503, 12504, 12517, 14931],
+    },
+    {"in": [], "out": []},
+)
 
-    def setUp(self):
-        model_dir = os.path.join(test_dir, "bigdata", "vecht_rekenen_tekenen_demo")
+NODES_EXPECTED = {
+    "1d": [11553, 12961],
+    "2d": [554, 555, 1466, 1467, 1480, 3894],
+    "2d_groundwater": [5928, 5929, 6840, 6841, 6854, 9268],
+}
+
+TIMESTEPS_EXPECTED = np.array(
+    [
+        0.0,
+        300.46375545,
+        602.91879098,
+        918.32741104,
+        1204.65442229,
+        1504.12043258,
+        1801.25668355,
+    ]
+)
+
+
+def helper_round_numpy(one_array):
```

Comment:
Die `helper_` prefix is wel lekker expliciet.

Vaak wordt daar ook kaal `_` voor gebruikt, daarmee zeg je ook "dit is iets interns voor deze module/class/enz".